src/convert/convertContext/recompositionFinalizer.ts

```diff
@@ -78,10 +78,27 @@ type ChildWithXml = {
 const recompose =
   (cache: XmlCache) =>
   async (stateValue: RecompositionStateValueWithParent): Promise<JsonMap> => {
-    await getXmlFromCache(cache)(stateValue.component);
+    const childComponents = stateValue.children?.toArray() ?? [];
+
+    // RecompositionState combines all labels metadata files into 1 component containing
+    // all the children.  This checks for multiple parent components and gets the xml
+    // file content from each.
+    if (stateValue.component.type.name === 'CustomLabels') {
```

Comment:
is there a Registry attribute (strategy, recomposition etc) we could use for this conditional instead of the type name?  

Thinking of the very soon decomposition stuff for Labels that we might not want running unintentionally here

Reply:
Great point!  I think there are probably 3 things that conditional needs:
1. the component has children:  i.e., `childComponents` is truthy
2. the component recomposition strategy is `startEmpty`
3. the component recomposition transformer is `nonDecomposed`